great_expectations/expectations/expectation.py

```diff
@@ -975,16 +972,49 @@ def run_diagnostics(
             raise_exceptions_for_backends=raise_exceptions_for_backends,
         )
 
+        backend_test_result_counts: List[
+            ExpectationBackendTestResultCounts
+        ] = ExpectationDiagnostics._get_backends_from_test_results(test_results)
+
+        renderers: List[
+            ExpectationRendererDiagnostics
+        ] = self._get_renderer_diagnostics(
+            expectation_type=description_diagnostics.snake_name,
+            test_diagnostics=test_results,
+            registered_renderers=_registered_renderers,
+        )
+
         maturity_checklist: ExpectationDiagnosticMaturityMessages = (
             self._get_maturity_checklist(
                 library_metadata=library_metadata,
                 description=description_diagnostics,
                 examples=examples,
                 tests=test_results,
+                backend_test_result_counts=backend_test_result_counts,
                 execution_engines=introspected_execution_engines,
             )
         )
 
+        # Set final maturity level based on status of all checks
+        all_experimental = all(
+            [check.passed for check in maturity_checklist.experimental]
+        )
+        all_beta = all([check.passed for check in maturity_checklist.beta])
+        all_production = all([check.passed for check in maturity_checklist.production])
+        if all_production and all_beta and all_experimental:
+            library_metadata.maturity = "PRODUCTION"
+        elif all_beta and all_experimental:
+            library_metadata.maturity = "BETA"
+        else:
+            library_metadata.maturity = "EXPERIMENTAL"
```

Comment:
Do we want to expose this maturity level in the printed checklist for users at some point?

Reply:
Just added a commit that does this.

```
Completeness checklist for ExpectColumnValuesToBeWeekday (BETA):
 ✔ Has a valid library_metadata object
 ✔ Has a docstring, including a one-line short description
    ✔ "Expect value to be weekday"
 ✔ Has at least one positive and negative example case, and all test cases pass
 ✔ Has core logic and passes tests on at least one Execution Engine
    ✔ All 2 tests for pandas are passing
 ✔ Passes all linting checks
 ✔ Has basic input validation and type checking
      Using default validate_configuration from template
 ✔ Has both statement Renderers: prescriptive and diagnostic
 ✔ Has core logic that passes tests for all applicable Execution Engines and SQL dialects
    ✔ All 2 tests for pandas are passing
   Has a full suite of tests, as determined by a code owner
   Has passed a manual review by a code owner for code standards and style guides
```